### PR TITLE
XMLFile loader added.

### PR DESCRIPTION
**Pull request by MaxUdk**
- accepts single or multiple XML sources (*.xml)
- accepts single or multiple XML schemas (*.xsd)
- both XML source and XML schema have to be passed over in the source list (regardless of order)
- if no schema can be found in the source list, the XMLFile loader for each XML source tries to identify every XML schema, it references to
- the XMLFile loader validates each XML source against all given XML schemas (*.xsd)
- if no schema can be found in the source list, the XMLFile loader validates each XML source against the referenced XML schemas
- converts to a dictionary of dictionaries, where d[XML source] contains a hierarchical dictionary representation of XML source's XML structure

### CODE REVIEW

**Comment by thorade** [comment]
IIRC, there once was an xml loader, but it was not used and removed.
@jraedler could you review and merge this PR?

File: CoTeTo/Loaders/XMLFile.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+# This file is part of CoTeTo - a code generation tool
+# 20170608 Joerg Raedler jraedler@udk-berlin.de
+#
+# XML file loader for CoTeTo
+# 20180208 Werner Kaul we.kaul@udk-berlin.de'
+#
+from CoTeTo.Loader import Loader
+from os.path import isfile
+
+from lxml import etree, objectify
```
Comment (thorade) [inline CoTeTo/Loaders/XMLFile.py:12]: This probably means lxml has to be added as a dependency in the setup.py I think. Or is it part of vanilla python, like the json and pickle reader?

File: Generators/TestXML/Templates/Main.tmpl [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#       Gen-Path: ${', '.join(systemCfg['generatorPath'])}
+#
+#   Generator
+#       Name:     ${generatorCfg['GENERATOR'].get('name')}
+#       Version:  ${generatorCfg['GENERATOR'].get('version')}
+
+<%! import json %>
+% for u,x in d.items():
+File: ${u}
+XML: ${json.dumps(x,indent=4)}
```
Comment (thorade) [inline Generators/TestXML/Templates/Main.tmpl:15]: is this correct?

**Comment by jraedler** [comment]
LXML is widely used but AFAIK not part of vanilla python. I would not introduce it as hard dependency for CoTeTo, but include the XMLFile loader anyway. When we move the lxml import from the top level to a setup() method in the loader, a missing lxml package will not break the import of the loader modules. Only the execution of the loader will fail. I'm making and testing the necessary changes in another branch and merge as soon as it's working.

**Comment by nytschgeusen** [comment]
@jraedler: can we merge the XML loader into the master branch ?

**Pull request merged**